From b49d83cf593641c00b9602f48275f847d665c887 Mon Sep 17 00:00:00 2001
From: Jiri Benc <jbenc@redhat.com>
Date: Fri, 12 Sep 2014 14:30:18 -0400
Subject: [net] openvswitch: Use IS_ERR_OR_NULL

Message-id: <867531689379af8e9c6b13d9765e34820e56877a.1410531783.git.jbenc@redhat.com>
Patchwork-id: 94006
O-Subject: [RHEL7.1 net PATCH 45/50] openvswitch: Use IS_ERR_OR_NULL
Bugzilla: 1110384
RH-Acked-by: Flavio Leitner <fbl@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110384
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7948291

commit d0e992aa0270663872c56f07473a7f43adee5bd5
Author: Himangi Saraogi <himangi774@gmail.com>
Date:   Sun Jul 27 12:37:46 2014 +0530

    openvswitch: Use IS_ERR_OR_NULL

    This patch introduces the use of the macro IS_ERR_OR_NULL in place of
    tests for NULL and IS_ERR.

    The following Coccinelle semantic patch was used for making the change:

    @@
    expression e;
    @@

    - e == NULL || IS_ERR(e)
    + IS_ERR_OR_NULL(e)
     || ...

    Signed-off-by: Himangi Saraogi <himangi774@gmail.com>
    Acked-by: Julia Lawall <julia.lawall@lip6.fr>
    Acked-by: Pravin B Shelar <pshelar@nicira.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/openvswitch/datapath.c b/net/openvswitch/datapath.c
index a32d28a..db8830e 100644
--- a/net/openvswitch/datapath.c
+++ b/net/openvswitch/datapath.c
@@ -783,7 +783,7 @@ static struct sk_buff *ovs_flow_cmd_build_info(const struct sw_flow *flow,
 
  skb = ovs_flow_cmd_alloc_info(ovsl_dereference(flow->sf_acts), info,
           always);
- if (!skb || IS_ERR(skb))
+ if (IS_ERR_OR_NULL(skb))
   return skb;
 
  retval = ovs_flow_cmd_fill_info(flow, dp_ifindex, skb,
-- 
1.7.1